src/main/scala/com/fulcrumgenomics/vcf/AssessPhasing.scala

```diff
@@ -0,0 +1,780 @@
+( is_truth: Boolean,
+  length: Long = 0,
+  count: Long = 0
+) extends Metric
+
+object PhasingUtils {
+  def getPhasingSetId(ctx: VariantContext): Int = {
+    ctx.getGenotype(0).getAttributeAsInt("PS", -1) // Integer.valueOf(ctx.getGenotype(0).getExtendedAttribute("PS", -1).toString)
+  }
+}
+
+private[vcf] class OverlapDetectorByContig(dict: SAMSequenceDictionary) {
+  import scala.collection.JavaConversions.collectionAsScalaIterable
+  private val detectors = dict.getSequences.map { seq =>
+    (seq.getSequenceName, new OverlapDetector[PhaseBlock](0, 0))
+  }.toMap
+
+  def addAll(iterator: Iterator[PhaseBlock]): this.type = {
+    iterator.foreach { block => this.detectors(block.getContig).addLhs(block, block) }
+    this
+  }
+
+  def overlapsAny(locatable: Locatable): Boolean = {
+    this.detectors(locatable.getContig).overlapsAny(locatable)
+  }
+
+  def getOverlaps(locatable: Locatable): Set[PhaseBlock] = {
+    import scala.collection.JavaConversions.asScalaSet
+    this.detectors(locatable.getContig).getOverlaps(locatable).toSet
+  }
+
+  def getAll: Iterator[PhaseBlock] = {
+    import scala.collection.JavaConversions.asScalaIterator
+    this.detectors.values.flatMap(_.getAll.iterator).toIterator
+  }
+}
+
+object PhaseBlock extends LazyLogging {
+  import scala.collection.mutable
+
+  /** Creates an overlap detector for blocks of phased variants.  Variants from the same block are found using the
+    * "PS" tag.
+    */
+  private[vcf] def toOverlapDetector(iterator: Iterator[VariantContext], dict: SAMSequenceDictionary): OverlapDetectorByContig = {
+    val detector = new OverlapDetectorByContig(dict=dict)
+    val progress = new ProgressLogger(logger)
+
+    logger.info("Creating the overlap detector")
+    // create the blocks
+    var curContig = ""
+    val phaseBlocks = mutable.HashMap[Int, PhaseBlock]()
+    while (iterator.hasNext) {
+      val ctx = iterator.next()
+      val phaseSetId: Int = PhasingUtils.getPhasingSetId(ctx)
+      if (phaseSetId > 0) {
+        if (ctx.getContig != curContig) {
+          detector.addAll(phaseBlocks.values.toIterator)
+          phaseBlocks.clear()
+          curContig = ctx.getContig
+        }
+        val phaseBlock = phaseBlocks.get(phaseSetId) match {
+          case Some(block) =>
+            require(block.getContig == ctx.getContig)
+            require(block.getStart <= ctx.getStart)
+            new PhaseBlock(contig=ctx.getContig, start=block.getStart, end=Math.max(block.getEnd, ctx.getEnd)) // extend the block
+          case None        =>
+            new PhaseBlock(contig=ctx.getContig, start=ctx.getStart, end=ctx.getEnd)
+        }
+        phaseBlocks.put(phaseSetId, phaseBlock)
+      }
+      progress.record(ctx.getContig, ctx.getStart)
+    }
+
+    // add any remaining
+    detector.addAll(phaseBlocks.values.toIterator)
+    phaseBlocks.clear()
+
+    detector
+  }
+}
+
+case class PhaseBlock private (contig: String, start: Int, end: Int) extends Locatable {
```

Comment:
I keep doing this in places too.  I think we should create a utility trait something like:
```scala
trait LocatableAdapter { // better name needed
  def contig: String
  def start: Int
  def end: Int
  def length: Int = end - start + 1

 def getContig: String = contig
 def getStart: Int = start
 def getEnd: Int = end
}
```

What do you think?